rclcpp/include/rclcpp/experimental/executors/events_executor/events_executor.hpp

```diff
@@ -253,6 +253,8 @@ class EventsExecutor : public rclcpp::Executor
   typename CollectionType::EntitySharedPtr
   retrieve_entity(typename CollectionType::Key entity_id, CollectionType & collection)
   {
+    std::lock_guard<std::recursive_mutex> lock(collection_mutex_);
```

Comment:
in this function scope, it is not obvious that it has always has an access to `current_entities_collection_`? according to the current implementation, it does always. but it could be leading unnecessary mutex lock if this function is used somewhere else w/o accessing `current_entities_collection_`?

Reply:
Done in [4f10321](https://github.com/ros2/rclcpp/pull/2187/commits/4f1032114df0620af51b44a0d93877d5538f5a25)